dmscripts/notify_suppliers_of_brief_withdrawal.py

```diff
@@ -0,0 +1,45 @@
+from dmutils.email.helpers import hash_string
+
+from dmscripts.helpers import env_helpers
+
+
+def get_brief_response_emails(data_api_client, brief_id):
+    responses = data_api_client.find_brief_responses(brief_id=brief_id, status="submitted").get("briefResponses")
+    return [response["respondToEmailAddress"] for response in responses]
+
+
+def create_context_for_brief(stage, brief):
+    return {
+        'brief_title': brief['title'],
+        'brief_link': '{0}/{1}/opportunities/{2}'.format(
+            env_helpers.get_web_url_from_stage(stage),
+            brief['frameworkFramework'],
+            brief['id']
+        )
+    }
+
+
+def main(data_api_client, mail_client, template_id, stage, logger, withdrawn_date=None, brief_id=None, dry_run=False):
+
+    withdrawn_briefs = data_api_client.find_briefs(withdrawn_on=withdrawn_date).get("briefs")
+
+    if brief_id:
+        withdrawn_briefs = filter(lambda i: i['id'] == brief_id, withdrawn_briefs)
+
+    for brief in withdrawn_briefs:
+        email_addresses = get_brief_response_emails(data_api_client, brief['id'])
+        if not email_addresses:
+            continue
```

Comment:
Do we gain much from breaking the loop here? Should we log if a withdrawn brief has no applications?

Reply:
| Do we gain much from breaking the loop here? 
No but I like how it reads, it explicitly informs the user that we take no action if we have no one to email.

| Should we log if a withdrawn brief has no applications?
I guess my thinking was that it _might_ come in useful to someone at some point but no one has asked for it and logging for a path that intentionally does nothing feels a bit weird

Reply:
OK, fair enough 😸